src/main/java/com/pubnub/api/models/consumer/push/payload/PushPayloadHelper.java

```diff
@@ -0,0 +1,268 @@
+package com.pubnub.api.models.consumer.push.payload;
+
+import com.pubnub.api.enums.PNPushEnvironment;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Setter
+@Accessors(chain = true)
+public class PushPayloadHelper {
+
+    private Map<String, Object> commonPayload;
+    private FCMPayload fcmPayload;
+    private MPNSPayload mpnsPayload;
+    private APNSPayload apnsPayload;
+
+    public Map<String, Object> build() {
+        Map<String, Object> payload = new HashMap<>();
+
+        if (apnsPayload != null) {
+            Map<String, Object> apnsMap = apnsPayload.toMap();
+            if (!apnsMap.isEmpty()) {
+                payload.put("pn_apns", apnsMap);
+            }
+        }
+
+        if (fcmPayload != null) {
+            Map<String, Object> fcmMap = fcmPayload.toMap();
+            if (!fcmMap.isEmpty()) {
+                payload.put("pn_gcm", fcmMap);
```

Comment:
Similarly do we need a separation for FCM/GCM?

Reply:
@davidnub 

No, they're the same. 

We still have a discrepancy in terms of GCM and FCM being used interchangeably in the SDK. But the logic and functionality remain the same.